pint/models/timing_model.py

```diff
@@ -477,15 +477,31 @@ def delay(self, toas, cutoff_component='', include_last=True):
             delay += df(toas, delay)
         return delay
 
-    def phase(self, toas):
+    def phase(self, toas, abs_phase=False):
         """Return the model-predicted pulse phase for the given TOAs."""
         # First compute the delays to "pulsar time"
         delay = self.delay(toas)
         phase = Phase(np.zeros(len(toas)) , np.zeros(len(toas)))
         # Then compute the relevant pulse phases
         for pf in self.phase_funcs:
             phase += Phase(pf(toas, delay))
-        return phase
+
+        # If the absolute phase flag is on, use the TZR parameters to compute
+        # the absolute phase.
+        if abs_phase:
+            if 'AbsPhase' not in list(self.component.keys()):
```

Comment:
Also, I loaded a par file using pint.models.get_model().  Even though it has TZR parameters, it is not adding an AbsPhase component.  Here is what is has:
```
In [3]: m.components
Out[3]: 
{'AstrometryEquatorial': <pint.models.astrometry.AstrometryEquatorial at 0x10ac87c50>,
 'Dispersion': <pint.models.dispersion_model.Dispersion at 0x10ac7bf10>,
 'SolarSystemShapiro': <pint.models.solar_system_shapiro.SolarSystemShapiro at 0x10ac96650>,
 'Spindown': <pint.models.spindown.Spindown at 0x10ac74450>}
```